cdap-ui/app/cdap/components/DataPrep/index.js

```diff
@@ -77,6 +79,17 @@ export default class DataPrep extends Component {
     this._isMounted = true;
     this.checkBackendUp(this.props);
     checkDataPrepHigherVersion();
+    this.storeSub = DataPrepStore.subscribe(() => {
+      let state = DataPrepStore.getState();
+      this.setState({ workspaceError: state.error.workspaceError }, () => {
```

Comment:
We don't need to set state if there is no workspaceError. Will avoid re-renders in UI.

Reply:
fixed